### PR TITLE
Add bokeh 0.11.1 from anaconda-recipes

### PR DESCRIPTION
**Pull request by ahmadia**

### CODE REVIEW

**Comment by jakirkham** [comment]
So, there is an existing `bokeh` recipe in PR ( https://github.com/conda-forge/staged-recipes/pull/898 ). However, please feel free to take over the effort. Feel free to take pieces from that and please make sure to add the listed maintainers from that recipe. Thanks.

**Comment by ahmadia** [comment]
Thanks @jakirkham - looks like there may be issues with the recipe test not completing due to different dependencies for the conda environment. Have you seen this before? CircleCI is just timing out :(

**Comment by ahmadia** [comment]
Looks like it was a spurious issue, I think this is going to be ready to merge in a few minutes.

**Comment by ahmadia** [comment]
Folks, this is ready for review. @birdsarah - you sent me on this goose chase, so you should probably review from Bokeh side.

File: recipes/bokeh/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    - bokeh.models
+    - bokeh.models.widgets
+    - bokeh.sampledata
+    - bokeh.server
+    - bokeh.sphinxext
+    - bokeh.util
+
+about:
+  home: http://bokeh.pydata.org/
+  license: BSD
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Please specify number of clauses.

File: recipes/bokeh/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    quickly and easily create interactive plots, dashboards, and data
+    applications.
+  doc_url: http://bokeh.pydata.org/en/latest/docs/user_guide.html
+  dev_url: http://github.com/bokeh/bokeh
+
+extra:
+    recipe-maintainers:
+        - almarklein
+        - birdsarah
+        - jakirkham
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: nit: please add terminal newline.

Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Also, please match spacing used elsewhere. So 2-space indents.

File: recipes/bokeh/run_test.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import sys
+import bokeh
+
+if sys.platform != 'win32':
+    bokeh.test()
```
Comment (jakirkham) [inline recipes/bokeh/run_test.py]: Why are we skipping tests on Windows?

File: recipes/bokeh/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+{% set version = "0.12.0" %}
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Is it alright if we start with `0.11.1`? I need that release for other reasons. We can bump the version to `0.12.0` in the feedstock.

File: recipes/bokeh/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Please add yourself here too, @ahmadia.

File: recipes/bokeh/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  name: bokeh
+  version: {{ version }}
+
+source:
+  fn: bokeh-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/b/bokeh/bokeh-{{ version }}.tar.gz
+  md5: 7c50e4a4afc42c240a04184587344078
+
+build:
+  number: 0
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Please add the following to the `build` section.

```
  script: python setup.py install
```

File: recipes/bokeh/build.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+$PYTHON setup.py install
```
Comment (jakirkham) [inline recipes/bokeh/build.sh]: Please remove this file as the `script` portion in `build` will replace it.

File: recipes/bokeh/bld.bat [hunk 7]
```diff
@@ -0,0 +1,2 @@
+"%PYTHON%" setup.py install
+if errorlevel 1 exit 1
```
Comment (jakirkham) [inline recipes/bokeh/bld.bat]: Please remove this file as the `script` portion in `build` will replace it.

**Comment by jakirkham** [comment]
Please drop the empty `LICENSE` file from the recipe as it is unused.

File: recipes/bokeh/run_test.py (hunk 3 above)
Comment (ahmadia) [inline recipes/bokeh/run_test.py]: don't know, this is inherited from anaconda-recipes, as the missing LICENSE file I forgot to add should indicate.

File: recipes/bokeh/meta.yaml (hunk 4 above)
Comment (ahmadia) [inline recipes/bokeh/meta.yaml]: Looks like the md5 is not automatically updated? I can set it back to 0.11.1

File: recipes/bokeh/meta.yaml (hunk 1 above)
Comment (ahmadia) [inline recipes/bokeh/meta.yaml]: addressed

File: recipes/bokeh/meta.yaml (hunk 2 above)
Comment (ahmadia) [inline recipes/bokeh/meta.yaml]: addressed, declined invitation to maintain :)

File: recipes/bokeh/meta.yaml (hunk 4 above)
Comment (ahmadia) [inline recipes/bokeh/meta.yaml]: addressed, set back to 0.11.1

File: recipes/bokeh/meta.yaml (hunk 5 above)
Comment (ahmadia) [inline recipes/bokeh/meta.yaml]: done

**Comment by ahmadia** [comment]
Alright, trying this again on 0.11.1

File: recipes/bokeh/run_test.py (hunk 3 above)
Comment (almarklein) [inline recipes/bokeh/run_test.py]: I think this is because until recently the tests would fail on Windows. This should not be necessary anymore though.

Comment (almarklein) [inline recipes/bokeh/run_test.py]: Correction, we still need this on 0.11.

File: recipes/bokeh/run_test.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+
+print('bokeh.__version__: %s' % bokeh.__version__)
+assert bokeh.__version__ == '0.12.0'
```
Comment (msarahan) [inline recipes/bokeh/run_test.py]: Version here needs to be 0.11.1.  Could also make it be os.environ['PKG_VERSION'] I think - but I don't totally remember if this variable is still defined at test time.

**Comment by ahmadia** [comment]
@jakirkham - can you take over this since you requested 0.11.1?

**Comment by ahmadia** [comment]
Needed to make some adjustments for all tests to pass.

**Comment by jakirkham** [comment]
Sure, I can try.

**Comment by jakirkham** [comment]
So, I'm a bit unclear on how adding the license works. This hasn't been something we have discussed previously. However, it does clash with some other use cases where recipes do have a license file (even though that is not the license of the recipe). We might need to think on this a bit if we are to keep it.

**Comment by jakirkham** [comment]
Please rename `LICENSE` to `RECIPE_LICENSE.txt`. Assuming this is what we decided, @conda-forge/core ?

File: recipes/bokeh/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    extend this capability with high-performance interactivity over very
+    large or streaming datasets. Bokeh can help anyone who would like to
+    quickly and easily create interactive plots, dashboards, and data
+    applications.
+  doc_url: http://bokeh.pydata.org/en/latest/docs/user_guide.html
+  dev_url: http://github.com/bokeh/bokeh
+
+extra:
+  channels:
+    - conda-forge
```
Comment (jakirkham) [inline recipes/bokeh/meta.yaml]: Please drop this.

File: recipes/bokeh/run_test.sh [hunk 10]
```diff
@@ -0,0 +1,3 @@
+ulimit -n 1024
+#{PYTHON} doesn't work
```
Comment (jakirkham) [inline recipes/bokeh/run_test.sh]: 😕

**Comment by mrocklin** [comment]
This recipe is very very old.  Having this on conda-forge is possibly counter-productive.

**Comment by mrocklin** [comment]
For some reason this is overriding the defaults channel.

```
Something is up with bokeh packages on conda-forge

    mrocklin@carbon:~$ conda create -n test-foo -c conda-forge python bokeh --dry-run
    Fetching package metadata ...........
    Solving package specifications: .
    
    Package plan for installation in environment /home/mrocklin/Software/anaconda/envs/test-foo:
    
    The following NEW packages will be INSTALLED:
    
        bokeh:            0.11.1-py36_0      conda-forge
        ca-certificates:  2017.1.23-0        conda-forge
        certifi:          2017.1.23-py36_0   conda-forge
        cycler:           0.10.0-py36_0      conda-forge
        dbus:             1.10.10-3          conda-forge
        expat:            2.1.0-2            conda-forge
        fontconfig:       2.12.1-4           conda-forge
        freetype:         2.7-1              conda-forge
        gettext:          0.19.7-1           conda-forge
        glib:             2.51.4-0           conda-forge
        gst-plugins-base: 1.8.0-0            conda-forge
        gstreamer:        1.8.0-1            conda-forge
        icu:              58.1-1             conda-forge
        jinja2:           2.9.5-py36_0       conda-forge
        jpeg:             9b-0               conda-forge
        libffi:           3.2.1-3            conda-forge
        libiconv:         1.14-4             conda-forge
        libpng:           1.6.28-0           conda-forge
        libxcb:           1.12-1             conda-forge
        libxml2:          2.9.4-4            conda-forge
        markupsafe:       0.23-py36_1        conda-forge
        matplotlib:       2.0.0-np112py36_3  conda-forge
        mkl:              2017.0.1-0                    
        ncurses:          5.9-10             conda-forge
        numpy:            1.12.1-py36_0                 
        openssl:          1.0.2h-3           conda-forge
        pandas:           0.19.2-np112py36_1 conda-forge
        pcre:             8.39-0             conda-forge
        pip:              9.0.1-py36_0       conda-forge
        pyparsing:        2.2.0-py36_0       conda-forge
        pyqt:             5.6.0-py36_0       conda-forge
        python:           3.6.1-0            conda-forge
        python-dateutil:  2.6.0-py36_0       conda-forge
        pytz:             2016.10-py36_0     conda-forge
        pyyaml:           3.12-py36_1        conda-forge
        qt:               5.6.2-1            conda-forge
        readline:         6.2-0              conda-forge
        requests:         2.13.0-py36_0      conda-forge
        setuptools:       33.1.1-py36_0      conda-forge
        sip:              4.18-py36_1        conda-forge
        six:              1.10.0-py36_1      conda-forge
        sqlite:           3.13.0-1           conda-forge
        tk:               8.5.19-1           conda-forge
        tornado:          4.4.2-py36_0       conda-forge
        wheel:            0.29.0-py36_0      conda-forge
        xorg-libxau:      1.0.8-3            conda-forge
        xorg-libxdmcp:    1.1.2-3            conda-forge
        xz:               5.2.2-0            conda-forge
        yaml:             0.1.6-0            conda-forge
        zlib:             1.2.11-0           conda-forge
```

**Comment by jakirkham** [comment]
Yeah, I have PRs open to update it. AppVeyor is just being really slow.

**Comment by mrocklin** [comment]
OK, thanks for taking this on.

**Comment by jakirkham** [comment]
NP. Sorry for the headache of having the old one pop-up.

**Pull request closed without merging**